src/components/FlowChart/Sidebar.tsx

```diff
@@ -0,0 +1,361 @@
+};
+
+const parseNode = (node: Node) => {
+  const properties = { ...node.properties };
+  const newProperties = {};
+  const label = node.label;
+
+  return { label, id: node.id, properties, newProperties };
+};
+
+function ReadOnly({ duplicateFlow }: { duplicateFlow: () => void }) {
+  return (
+    <div className="readonly-overlay py-4 px-4 text-center">
+      <h4 className="my-4">
+        <b>
+          This flow is <span className="text-muted">readonly</span>
+        </b>
+      </h4>
+      <p className="my-4">To edit this flow, please duplicate it.</p>
+      <Button onClick={duplicateFlow}>Duplicate Flow</Button>
+    </div>
+  );
+}
+
+type EditLinkProps = {
+  link: Link;
+  nodes: NodesObject;
+  updateLink: (linkId: string, nodeFrom: string, nodeTo: string) => void;
+  deleteSelection: () => void;
+};
+
+function EditLink({ link, nodes, updateLink, deleteSelection }: EditLinkProps) {
+  const nodeFrom = nodes[link.from.nodeId];
+  const nodeTo = nodes[link.to.nodeId];
+
+  let choices = Object.keys(nodes).map((id) => {
+    return { label: nodes[id].label || nodes[id].properties.name, id };
+  });
+
+  return (
+    <div className="h-100 d-flex flex-column">
+      <h5 className="px-3 py-2 mb-0 border-bottom">
+        <b>Edit Connection</b>
+      </h5>
+      <div className="flex-fill px-2">
+        <div className="p-2 mb-1">
+          <p className="mb-1">
+            <b>From</b>
+          </p>
+          <FormControl
+            className="mb-2"
+            as="select"
+            onChange={(e) => updateLink(link.id, e.target.value, nodeTo.id)}
+            value={nodeFrom.id}
+          >
+            {choices.map((choice) => (
+              <option key={choice.id} value={choice.id}>
+                {choice.label}
+              </option>
+            ))}
+          </FormControl>
+          <p className="mb-1">
+            <b>To</b>
+          </p>
+          <FormControl
+            className="mb-2"
+            as="select"
+            onChange={(e) => updateLink(link.id, nodeFrom.id, e.target.value)}
+            value={nodeTo.id}
+          >
+            {choices.map((choice) => (
+              <option key={choice.id} value={choice.id}>
+                {choice.label}
+              </option>
+            ))}
+          </FormControl>
+        </div>
+      </div>
+      <div className="p-2">
+        <Button variant="danger" className="w-100" onClick={deleteSelection}>
+          Delete Connection
+        </Button>
+      </div>
+    </div>
+  );
+}
+
+type EditNodeProps = {
+  node: Node;
+  updateLabel: (label: string) => void;
+  updateNewValue: (key: string, value: any) => void;
+  updateExistingValue: (key: string, value: any) => void;
+  deleteSelection: () => void;
+};
+
+function EditNode({
+  node,
+  updateLabel,
+  updateNewValue,
+  updateExistingValue,
+  deleteSelection,
+}: EditNodeProps) {
+  const availableProperties = Store.getAvailableProperties();
```

Comment:
This will run on every render. Just try to only connect the top level component otherwise we will stay too tightly coupled to the global Store..